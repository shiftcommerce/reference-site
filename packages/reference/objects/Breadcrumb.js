// Libraries
import { Component } from 'react'
import Link from 'next/link'

class Breadcrumb extends Component {
  constructor (props) {
    super(props)
    this.renderBreadcrumbs = this.renderBreadcrumbs.bind(this)
  }

  renderBreadcrumbs (trail) {
    return (
      trail && trail.map((crumb, idx) =>
        <li key={idx} className='o-breadcrumb__item'>
          <Link href={crumb.canonical_path} >
            <a className='o-breadcrumb__link'>{ crumb.title }</a>
          </Link>
        </li>
      )
    )
  }

  render () {
    let {
      trail,
      ...otherProps
    } = this.props

    return (
      <nav>
        <ol className={'o-breadcrumb'} {...otherProps}>
          <li className='o-breadcrumb__item'>
            <Link href='/home/index' as='/'>
              <a className='o-breadcrumb__link'>Home</a>
            </Link>
          </li>

          { this.renderBreadcrumbs(trail) }
        </ol>
      </nav>
    )
  }
}

export default Breadcrumb
