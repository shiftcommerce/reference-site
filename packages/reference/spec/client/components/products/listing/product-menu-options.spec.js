// Component
import ProductMenuOptions from '../../../../../client/components/products/listing/product-menu-options'
import ProductMenuDropdowns from '../../../../../client/components/products/listing/product-menu-dropdowns'

// Objects
import Breadcrumb from '../../../../../client/objects/breadcrumb'

test('renders correctly', () => {
  // Arrange
  const initialProps = {
    toggleFiltering: () => (null),
    filterCount: 3
  }

  // Act
  const wrapper = shallow(
    <ProductMenuOptions {...initialProps} />
  )

  const rootTree = wrapper.find('.c-product-listing__menu-options')

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(rootTree.find(Breadcrumb))
  expect(rootTree.find(ProductMenuDropdowns))
})
