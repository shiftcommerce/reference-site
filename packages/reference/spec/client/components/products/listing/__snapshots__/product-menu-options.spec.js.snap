// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="c-product-listing__menu-options"
>
  <div
    className="c-product-listing__menu-options-filters"
  >
    <h2
      className="c-product-listing__menu-options-filters-title"
    >
      Filters
    </h2>
    <div
      className="c-product-listing__menu-options-filters-applied"
    >
      <AlgoliaCurrentRefinements(UnknownComponent) />
    </div>
    <Button
      className="c-product-listing__menu-options-filters-button"
      onClick={[Function]}
    />
  </div>
  <div
    className="c-product-listing__menu-options-sort-by"
  >
    <h2
      className="c-product-listing__menu-options-sort-by-title"
    >
      Sort by:
    </h2>
    <AlgoliaSortBy(SortBy)
      defaultRefinement="instant_search"
      items={
        Array [
          Object {
            "label": "Featured",
            "value": "instant_search",
          },
          Object {
            "label": "Rating asc.",
            "value": "instant_search_product_rating_asc",
          },
          Object {
            "label": "Rating desc.",
            "value": "instant_search_product_rating_desc",
          },
        ]
      }
    />
    <Button
      className="c-product-listing__menu-options-sort-by-button u-hidden-d"
    />
  </div>
</div>
`;
